sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs

```diff
@@ -122,19 +122,19 @@ public ProvisionableTemplateAttribute(string resourceName) { }
 }
 namespace System.ClientModel
 {
-    public abstract partial class RefreshableToken : System.ClientModel.Token
+    public partial class AccessToken
     {
-        public RefreshableToken(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) : base (default(string), default(string), default(System.DateTimeOffset), default(System.DateTimeOffset?)) { }
-        public abstract System.Threading.Tasks.Task RefreshAsync(System.Threading.CancellationToken cancellationToken);
-    }
-    public partial class Token
-    {
-        public Token(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) { }
+        public AccessToken(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) { }
         public System.DateTimeOffset ExpiresOn { get { throw null; } protected set { } }
         public System.DateTimeOffset? RefreshOn { get { throw null; } protected set { } }
         public string TokenType { get { throw null; } protected set { } }
         public string TokenValue { get { throw null; } protected set { } }
     }
+    public abstract partial class RefreshableToken : System.ClientModel.AccessToken
+    {
+        public RefreshableToken(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) : base (default(string), default(string), default(System.DateTimeOffset), default(System.DateTimeOffset?)) { }
+        public abstract System.Threading.Tasks.Task RefreshAsync(System.Threading.CancellationToken cancellationToken);
```

Comment:
I thought we decided the token needs to be immutable?

Reply:
I was going to make another pass for the token changes. But this brings up a reason why we should consider having the TokenProvider type return an interface like IToken and then we can make these types structs to deal with the immutability while not preventing future expansion without an ever-growing property list like we have in Core today. Thoughts?

Reply:
You can have immutable classes.